File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/75-SortColors/75-SortColors.java
// Last updated: 8/4/2026, 7:23:24 PM
class Solution {
    public void sortColors(int[] nums) {
        //dutch national flag algo

        int i = 0 , j = 0 , k = nums.length - 1;

        while(j <= k){
            if(nums[j] == 0){
                int temp = nums[i];
                nums[i] = nums[j];
                nums[j] = temp;
                j++;
                i++;
            }
            else if(nums[j] == 2){
                int temp = nums[j];
                nums[j] = nums[k];
                nums[k] = temp;
                k--;
            }
            else{
                j++;
            }
        }
    }
}

File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/75-SortColors/75-SortColors.cpp
// Last updated: 8/4/2026, 7:23:18 PM
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int n = nums.size();
        int l = 0 , mid = 0 , r = n-1;

        while(mid <= r){
            if(nums[mid] == 0){
                swap(nums[l] , nums[mid]);
                l++;
                mid++;
            }
            else if(nums[mid] == 1){
                mid++;
            }
            else{
                swap(nums[r] , nums[mid]);
                r--;
            }

        }
    }
};